src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -402,6 +411,117 @@ def is_ne_cell(self, other) -> bool:
         # this does not hold in general, but here we use finite sets
         return not self.is_equal_cell(other)
 
+    def get_state(self):
+        if isinstance(self, AtomicCell):
+            return self.atoms_state
+        elif hasattr(self, 'particles_state'):
+            return self.particles_state
+        else:
+            raise AttributeError(
+                'The class does not have atoms_state or particles_state'
+            )
+
+    def get(self, key, logger=None):
```

Comment:
This is the crux of the implementation of getting the labels from either type of cell. Still not sure if there is a better way

Reply:
I think we do need to be able to differentiate at various points, e.g., to selectively skip certain normalizations. I've been trying to come up with a more elegant way than constantly looking for the cell type, but I'm not positive there is one at this point.